pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -1255,7 +1259,7 @@ func findProbe(probes []network.Probe, probe network.Probe) bool {
 
 func findRule(rules []network.LoadBalancingRule, rule network.LoadBalancingRule) bool {
 	for _, existingRule := range rules {
-		if strings.EqualFold(*existingRule.Name, *rule.Name) {
+		if strings.EqualFold(*existingRule.Name, *rule.Name) && reflect.DeepEqual(existingRule.LoadBalancingRulePropertiesFormat, rule.LoadBalancingRulePropertiesFormat) {
```

Comment:
If rule name equals, but not deep equal, we should replace the current rule. With this logic, it will add a new one, needs update?

Reply:
it actually replace the rule in such case by removing old one from list and append a new to the list.